Cascading: java flight server + relay

cdoron opened the pull request:
initial version
resolves #16

Signed-off-by: Doron Chen <cdoron@il.ibm.com>

cdoron committed:
Cascading: java flight server + relay
initial version

Signed-off-by: Doron Chen <cdoron@il.ibm.com>

simanadler commented on Java-FlightServer/src/flight-client/main/java/flight_client/org/m4d/adp/flight_client/RunBenchmark.java:
```diff
@@ -0,0 +1,152 @@
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+public class RunBenchmark {
```
Please add comments describing what it is you are testing and how you are testing it.  In general some documentation for all the functions would be helpful

simanadler commented on Java-FlightServer/src/flight-client/main/java/flight_client/org/m4d/adp/flight_client/MyFlightClient.java:
```diff
@@ -0,0 +1,55 @@
+
+import java.nio.charset.StandardCharsets;
+
+public class MyFlightClient {
```
Please add some comments describing what this client does.

simanadler commented on Java-FlightServer/src/flight-server/main/java/org/m4d/adp/flight_server/ExampleProducer.java:
```diff
@@ -0,0 +1,104 @@
+    private final VectorSchemaRoot constVectorSchemaRoot;
+    private boolean isNonBlocking = false;
+
+    public ExampleProducer(Location location, BufferAllocator allocator) {
```
Can you please add documentation here and throughout this file?

simanadler commented on Java-FlightServer/src/flight-server/main/java/org/m4d/adp/flight_server/RelayProducer.java:
```diff
@@ -0,0 +1,132 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class RelayProducer extends NoOpFlightProducer {
```
Again, please document

simanadler reviewed (commented):
Thanks for this initial PR @cdoron! Please add to the PR a link to the issue and/or document describing the problem or feature you are addressing.  You mention that this is an initial version.  Please indicate what parts have and have not been addressed by this PR. Code needs documentation as well.    FYI - I did not review the code in depth.

cdoron committed:
rearrange directories

Signed-off-by: Doron Chen <cdoron@il.ibm.com>

cdoron committed:
remove main/java directories

Signed-off-by: Doron Chen <cdoron@il.ibm.com>

the-mesh-for-data commented:
Addresses Issue #16

cdoron commented:
addresses issue #16

Commit:
removed backpressure handling
This is improves performance, and
does not currently cause crashes.
In the future will try to add backpressure
handling without busy wait

Signed-off-by: CDORON@il.ibm.com <cdoron@lnx-tes.sl.cloud9.ibm.com>

the-mesh-for-data commented:
Hi @simanadler : I added comments in a separate PR.
This PR contains the code for a flight server which contains a single dataset in memory, a relay flight server which gets a record-batch and transmits a record batch (either identical or after a transformation) and client code.
I labeled the PR 'initial' since there is still further investigations to be done on how to improve performance of the relay server. Also, we currently support only a single hard-coded transformation. We still need to determine the proper design for supporting multiple transformations.

cdoron commented:
Hi @simanadler : I added comments in a separate PR.
This PR contains the code for a flight server which contains a single dataset in memory, a relay flight server which gets a record-batch and transmits a record batch (either identical or after a transformation) and client code.
I labeled the PR 'initial' since there is still further investigations to be done on how to improve performance of the relay server. Also, we currently support only a single hard-coded transformation. We still need to determine the proper design for supporting multiple transformations.
